cmake/src.cmake

```diff
@@ -0,0 +1,87 @@
+configure_file(
+    "${PROJECT_SOURCE_DIR}/cmake/matio_pubconf.cmake.in"
+    "${CMAKE_CURRENT_BINARY_DIR}/src/matio_pubconf.h"
+    ESCAPE_QUOTES @ONLY
+)
+
+configure_file(
+    "${PROJECT_SOURCE_DIR}/cmake/matioConfig.cmake.in"
+    "${CMAKE_CURRENT_BINARY_DIR}/src/matioConfig.h"
+    ESCAPE_QUOTES @ONLY
+)
+
+set(MATIO_SOURCES
+    ${PROJECT_SOURCE_DIR}/src/endian.c
+    ${PROJECT_SOURCE_DIR}/src/mat.c
+    ${PROJECT_SOURCE_DIR}/src/io.c
+    ${PROJECT_SOURCE_DIR}/src/inflate.c
+    ${PROJECT_SOURCE_DIR}/src/mat73.c
+    ${PROJECT_SOURCE_DIR}/src/matvar_cell.c
+    ${PROJECT_SOURCE_DIR}/src/matvar_struct.c
+    ${PROJECT_SOURCE_DIR}/src/mat4.c
+    ${PROJECT_SOURCE_DIR}/src/mat5.c
+    ${PROJECT_SOURCE_DIR}/src/snprintf.c
+    ${PROJECT_SOURCE_DIR}/src/read_data.c
+    ${PROJECT_SOURCE_DIR}/src/mat5.h
+    ${PROJECT_SOURCE_DIR}/src/mat73.h
+    ${PROJECT_SOURCE_DIR}/src/matio_private.h
+    ${PROJECT_SOURCE_DIR}/src/mat4.h
+    ${PROJECT_SOURCE_DIR}/src/matio.h
+    ${PROJECT_SOURCE_DIR}/visual_studio/matio.rc
+    ${PROJECT_SOURCE_DIR}/visual_studio/matio.def
+    ${CMAKE_CURRENT_BINARY_DIR}/src/matio_pubconf.h
+    ${CMAKE_CURRENT_BINARY_DIR}/src/matioConfig.h
+)
+
+if (MATIO_SHARED)
+    add_library(matio SHARED ${MATIO_SOURCES})
+else()
+    add_library(matio STATIC ${MATIO_SOURCES})
+endif()
+add_library(matio::matio ALIAS matio)
+
+target_include_directories(matio
+    INTERFACE ${PROJECT_SOURCE_DIR}/src
+    PUBLIC ${CMAKE_CURRENT_BINARY_DIR}/src
+)
+if(WIN32 AND NOT MSVC)
+    target_compile_definitions(matio PUBLIC GCC_WINDRES)
```

Comment:
This does not seem to work when checking Cygwin build for the resource compilation of matio.rc.